src/Checks/DatabaseConnectionCountCheck.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Appkeep\Laravel\Checks;
+
+use Exception;
+use Appkeep\Laravel\Check;
+use Appkeep\Laravel\Result;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Cache;
+use Appkeep\Laravel\Database\MySqlInspector;
+use Appkeep\Laravel\Database\PostgresInspector;
+use Appkeep\Laravel\Database\SqlServerInspector;
+
+class DatabaseConnectionCountCheck extends Check
+{
+    protected $warnAt;
+
+    protected $failAt;
+
+    private $connection;
+
+    private $inspectors = [
+        'mysql' => MySqlInspector::class,
+        'sqlsrv' => SqlServerInspector::class,
+        'postgres' => PostgresInspector::class,
+    ];
+
+    /**
+     * Check a different connection
+     */
+    public function connection($connection)
+    {
+        $this->connection = $connection;
+
+        return $this;
+    }
+
+    public function warnIfConnectionCountIsAbove($count)
+    {
+        $this->warnAt = (int) $count;
+
+        return $this;
+    }
+
+    public function failIfConnectionCountIsAbove($count)
+    {
+        $this->failAt = (int) $count;
+
+        return $this;
+    }
+
+    /**
+     * @var Result
+     */
+    public function run()
+    {
+        $connection = $this->connection ?? config('database.default');
+        $meta = ['connection' => $connection];
+
+        $db = DB::connection($connection);
+        $driver = $db->getDriverName();
+
+        // If there's no inspector for this connection, we can't check it.
+        // In that case, we'll just skip this check.
+        if (! isset($this->inspectors[$driver])) {
+            return Result::ok()->meta($meta);
+        }
+
+        /**
+         * @var DatabaseIn
+         */
+        $inspector = app($this->inspectors[$driver], [
+            'connection' => $db,
+        ]);
+
+        $cacheKey = 'appkeep.db.' . $connection . '.connection_count';
+
+        $maxConnections = Cache::remember(
+            $cacheKey,
+            now()->addHours(2),
+            function () use ($inspector) {
+                try {
+                    return $inspector->maximumConnectionCount();
+                } catch (Exception $e) {
+                    dd($e);
+                    // Failed to get the maximum connection count.
+                    report($e);
+
+                    // Let's just assume the default.
+                    return 100;
+                }
+            }
+        );
```

Comment:
Doesn't this (Cache::remember) cache the default max connections value (100) ? Shouldn't we instead check the actual maximum connection count in the next run instead of getting the default value result from cache ?

Reply:
Good that you pointed this out. Looks like I forgot to remove that `dd`. About the default max connections, it should kick in if the app's db user does not have the privileges to query the max connection count. If the DB connection fails altogether, I assume the code will crash earlier (which is ok, as any unhandled exception in a check run is already wrapped in a try/catch where it gets invoked).